pkg/clusteragent/admission/mutate/cws_instrumentation_test.go

```diff
@@ -886,3 +887,48 @@ func Test_initCWSInitContainerResources(t *testing.T) {
 		})
 	}
 }
+
+func Test_injectCWSWorkloadTags(t *testing.T) {
+	mockConfig := config.Mock(t)
+	mockConfig.SetWithoutSource("admission_controller.cws_instrumentation.inject_workload_tags", true)
+	mockConfig.SetWithoutSource("admission_controller.mutate_unlabelled", true)
+
+	newPod := func() *corev1.Pod {
+		pod := fakePod("foo-pod")
+		pod.Namespace = "foo-ns"
+		pod.Spec.Containers[0].Image = "myworkload:8.3.2"
+		return pod
+	}
+	tests := []struct {
+		name        string
+		labels      map[string]string
+		pod         *corev1.Pod
+		wantPodFunc func() corev1.Pod
+		found       bool
+		injected    bool
+	}{
+		{
+			name:   "nominal case",
+			labels: map[string]string{"tags.datadoghq.com/service": "dd-agent", "tags.datadoghq.com/version": "7"},
+			pod:    newPod(),
+			wantPodFunc: func() corev1.Pod {
+				pod := fakePod("foo-pod")
+				pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, fakeEnvWithValue("DD_WORKLOAD_IMAGE_NAME", "myworkload"))
+				pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, fakeEnvWithValue("DD_WORKLOAD_IMAGE_TAG", "8.3.2"))
+				return *pod
+			},
+			found:    true,
+			injected: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			instrumentation := CWSInstrumentation{}
+			err := instrumentation.injectCWSWorkloadTags(tt.pod, "", nil)
+			assert.Equal(t, err, nil)
```

Comment:
Since err would break the rest of the test, should this be:
```suggestion
			require.Equal(t, err, nil)
```